task2.js

```diff
@@ -0,0 +1,36 @@
+// write a constructor function `WorkingPersonFunc` which inherits from `PersonFunc` and implements additional method:
+
+// - getProfessionalNameAndRank() => returns string in format `"[FirstName] [LastName], [job title], job experience [job experience duration] years"`, e.g. _"User Name, Front-end developer, job experience 5 years"_.
+// // NOTE: Please implement error handling in case if some fields are not available.
+
+const { PersonFunc } = require("./task1.js");
+const { SarahMay, JeremieBrown, Elizabeth } = require("./constants.js");
+const { validateFields } = require("./utils.js");
+
+function WorkingPersonFunc({ firstName, lastName, birthDate, address, job }) {
```

Comment:
after rewrite all methods from task1 to `Prototype Inheritance` here u should `Set up prototype inheritance` and `Reset the constructor reference` and then create new method `WorkingPersonFunc.prototype.getProfessionalNameAndRank`